### PR TITLE
Add prospective-parachain subsystem to minimal-relay-node + QoL improvements

### PR DESCRIPTION
**Pull request by skunert**
This PR contains some fixes and cleanups for parachain nodes:

1. When using async backing, node no longer complains about being unable to reach the prospective-parachain subsystem.
2. Parachain warp sync now informs users that the finalized para block has been retrieved.
```
2023-11-08 13:24:42 [Parachain] 🎉 Received finalized parachain header #5747719 (0xa0aa…674b) from the relay chain.
```
3. When a user supplied an invalid `--relay-chain-rpc-url`, we were crashing with a very verbose message. Removed the `expect` and improved the error message.
```
2023-11-08 13:57:56 [Parachain] No valid RPC url found. Stopping RPC worker.
2023-11-08 13:57:56 [Parachain] Essential task `relay-chain-rpc-worker` failed. Shutting down service.
Error: Service(Application(WorkerCommunicationError("RPC worker channel closed. This can hint and connectivity issues with the supplied RPC endpoints. Message: oneshot canceled")))
```

### CODE REVIEW

File: cumulus/client/service/src/lib.rs [hunk 1]
```diff
@@ -560,11 +560,15 @@ where
...
+			let finalized_header = B::Header::decode(&mut &validation_data.parent_head.0[..])
 				.map_err(|e| format!("Failed to decode parachain head: {e}"))?;
 
-			log::debug!(target: LOG_TARGET_SYNC, "Target block reached {:?}", target_block);
-			let _ = sender.send(target_block);
+			log::info!(
+				"🎉 Received finalized parachain header #{} ({}) from the relay chain.",
+				finalized_header.number(),
+				finalized_header.hash()
+			);
```
Comment (bkchr) [inline cumulus/client/service/src/lib.rs:570]: The syncing logs already inform us about the progress?

Comment (skunert) [reply cumulus/client/service/src/lib.rs:570]: Yeah it is not an absolute necessity, you already get the occasional `⏩ Waiting for pending target block (7 peers), best: #0 (0x4823…771a), finalized #0 (0x4823…771a)`. However this here is tells you which block is the new target and is immediately printed when we retrieve the finalized one. I am also against spamming logs, but in my opinion this additional info does not hurt.

**Review by bkchr — APPROVED** [review]

File: cumulus/client/service/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply cumulus/client/service/src/lib.rs:570]: But then maybe change the text to `Received target parachain header ..` or something like this.

**Review by davxy — APPROVED** [review]

File: cumulus/client/service/src/lib.rs [hunk 2]
```diff
@@ -560,11 +560,15 @@ where
...
 				.ok_or("Could not find parachain head in relay chain")?;
 
-			let target_block = B::Header::decode(&mut &validation_data.parent_head.0[..])
+			let finalized_header = B::Header::decode(&mut &validation_data.parent_head.0[..])
 				.map_err(|e| format!("Failed to decode parachain head: {e}"))?;
 
-			log::debug!(target: LOG_TARGET_SYNC, "Target block reached {:?}", target_block);
-			let _ = sender.send(target_block);
+			log::info!(
+				"🎉 Received finalized parachain header #{} ({}) from the relay chain.",
```
Comment (skunert) [inline cumulus/client/service/src/lib.rs:567]:
```suggestion
				"🎉 Received target parachain header #{} ({}) from the relay chain.",
```

**Pull request merged**